libs/next_gen_ui_agent/component_selection_common.py

```diff
@@ -0,0 +1,212 @@
+    UI_COMPONENTS_DESCRIPTION_SUPPORTED
+    + """
+* table - component to visualize array of objects with multiple items (typically 3 or more) in a tabular format. Use when user explicitly requests a table, or for data with many items (especially >6), small number of fields, and short values.
+* set-of-cards - component to visualize array of objects with multiple items. Use for data with fewer items (<6), high number of fields, or fields with long values. Also good when visual separation between items is important.
+""".strip()
+)
+
+
+def get_ui_components_description(unsupported_components: bool) -> str:
+    """Get UI components description for system prompt based on the unsupported_components flag."""
+    if unsupported_components:
+        return UI_COMPONENTS_DESCRIPTION_ALL
+    else:
+        return UI_COMPONENTS_DESCRIPTION_SUPPORTED
+
+
+# ============================================================================
+# SECTION 2: Internal Building Blocks (Private - not imported by strategies)
+# ============================================================================
+_COMMON_RULES = """RULES:
+- If user explicitly requests a component type ("table", "chart", "cards"), USE IT unless data structure prevents it
+- Select one component in "component" field
+- Provide "title", "reasonForTheComponentSelection", "confidenceScore" (percentage)"""
+
+_COMMON_FIELD_SELECTION_RULES = """- Select relevant Data fields based on User query
+- Each field must have "name" and "data_path"
+- Do not use formatting or calculations in "data_path"
+"""
+
+_JSONPATH_REQUIREMENTS = """JSONPATH REQUIREMENTS:
+- Analyze actual Data structure carefully
+- If fields are nested (e.g., items[*].movie.title), include full path
+- Do NOT skip intermediate objects
+- Use [*] for array access"""
+
+# ============================================================================
+# SECTION 3: One-Step Strategy (Public API)
+# ============================================================================
+ONESTEP_PROMPT_RULES = f"""{_COMMON_RULES}
+- Generate JSON only
+{_COMMON_FIELD_SELECTION_RULES}
+
+{_JSONPATH_REQUIREMENTS}"""
+
+ONESTEP_RESPONSE_EXAMPLES = """Response example for multi-item data:
+{
+    "title": "Orders",
+    "reasonForTheComponentSelection": "User explicitly requested a table, and data has multiple items with short field values",
+    "confidenceScore": "95%",
+    "component": "table",
+    "fields" : [
+        {"name":"Name","data_path":"orders[*].name"},
+        {"name":"Creation Date","data_path":"orders[*].creationDate"}
+    ]
+}
+
+Response example for one-item data:
+{
+    "title": "Order CA565",
+    "reasonForTheComponentSelection": "One item available in the data",
+    "confidenceScore": "75%",
+    "component": "one-card",
+    "fields" : [
+        {"name":"Name","data_path":"order.name"},
+        {"name":"Creation Date","data_path":"order.creationDate"}
+    ]
+}
+
+Response example for bar chart:
```

Comment:
Seems system prompt is much larger with all the additions. I'm interested in performance impact of this, should be visible from evals.